nginx_proxy: Update documentation (#3373)
home-assistant/addons: nginx_proxy/DOCS.md

## How to use

> The NGINX Proxy add-on is commonly used in conjunction with the [Duck DNS](https://github.com/home-assistant/addons/tree/master/duckdns) add-on to set up secure remote access to your Home Assistant instance. The following instructions covers this scenario. Another option for certificate management is [Let's Encrypt](https://github.com/home-assistant/addons/tree/master/letsencrypt) add-on.

Reviewer: ```suggestion
The NGINX Proxy add-on is commonly used in conjunction with the [Duck DNS](https://github.com/home-assistant/addons/tree/master/duckdns) and/or the [Let's Encrypt](https://github.com/home-assistant/addons/tree/master/letsencrypt) add-on to set up secure remote access to your Home Assistant instance. The following instructions covers this scenario.
```

1. The certificate to your registered domain should already be created via [Duck DNS](https://github.com/home-assistant/addons/tree/master/duckdns), [Let's Encrypt](https://github.com/home-assistant/addons/tree/master/letsencrypt) or another method. Make sure that the certificate files exist in the `/ssl` directory.

> 2. You must add the following section to your [Home Assistant global configuration.yaml](https://www.home-assistant.io/docs/configuration/), otherwise the Home Assistant server process will respond with `400 Bad Request` to requests arriving via Nginx add-on reverse proxy.

Reviewer: I think we should still tell users to remove any HA specific SSL options in this step, so:

```suggestion
2. You must add the following section to your [Home Assistant configuration.yaml](https://www.home-assistant.io/docs/configuration/), otherwise the Home Assistant Core process will respond with `400 Bad Request` to requests arriving via this NGINX Porxy add-on. If the `http` section is using the `ssl_certificate`, `ssl_key` or `server_port` keys, make sure to remove them.
```

The question is if is really worthwhile to document what happens when not following this process? It might be more confusing? :thinking:

Author: I intentionally included the 400 response and why you get that so that people seeing that would understand what is happening. I think it is a good idea to include the 400 so you can go "oh, that's where I am, here is the fix I am looking for", the note about HA core is not so valuable.

How long do you think the note about the `http` sections needs to be there? I moved it to the bottom so that it might eventually go away, as a fresh installation like mine does not contain that and I I have no idea how prevalent that is.

Your suggestion is 👍 as it stands, just that there's a typo in `Porxy`.

Reviewer: Whoops, fixed the "Porxy" typo.

> I think it is a good idea to include the 400 so you can go "oh, that's where I am, here is the fix I am looking for", the note about HA core is not so valuable.

I guess you could miss any one of these numbers, and you'll run into another issue. IMHO, these type of what happen if you did not follow on every step would hurt the flow of the instructions very much. I'd prefer to have them in a Troubleshooting section further down.

> How long do you think the note about the http sections needs to be there?

Yeah good question not sure. These are still valid options, and people might migrate from a non-Proxy setup to a Proxy setup. This could also be the case for new installations, e.g. if people followed instructions using Core SSL implementation.

   ```yaml
   http:
     use_x_forwarded_for: true
     trusted_proxies:
       - 172.30.33.0/24
   ```
3. In the nginx addon configuration, change the `domain` option to the domain name you registered (from DuckDNS or any other domain you control).
4. Leave all other options as-is.
5. Save configuration.
6. Start the add-on.
7. Have some patience and wait a couple of minutes.
8. Check the add-on log output to see the result.

> 
> Note: some options in the `http:` section of `configuration.yaml` are no longer necessary for this scenario, and should be commented out or removed:
>    - `ssl_certificate`
>    - `ssl_key`
>    - `server_port`

Reviewer: ```suggestion
```
